bundler.py

```diff
@@ -0,0 +1,219 @@
+# SETUP
+
+def start():
+    config = L2Config()
+    setup_4337_contracts(config)
+    setup_stackup_bundler(config)
+    setup_paymaster(config)
+
+# --------------------------------------------------------------------------------------------------
+
+def setup_4337_contracts(config: L2Config):
+    github_url = "https://github.com/0xFableOrg/account-abstraction.git"
+
+    if os.path.isfile("account-abstraction"):
+        raise Exception("Error: 'account-abstraction' exists as a file and not a directory.")
+    elif not os.path.exists("account-abstraction"):
+        lib.clone_repo(github_url, "clone the account-abstraction repository")
+
+    # If contracts have not been previously deployed
+    if not os.path.exists("account-abstraction/deployments/opstack"):
+        log_file = "logs/build_4337_contracts.log"
+        lib.run_roll_log(
+            "install account abstraction dependencies", 
+            command=deps.cmd_with_node("yarn install"), 
+            cwd="account-abstraction",
+            log_file=log_file
+        )
+        # set private key for deployment
+        if config.deployer_key is None:
+            priv_key = input("Enter private key that you would like to deploy contracts with: ")
+            config.deployer_key = priv_key
+        else:
+            priv_key = config.deployer_key
+        lib.run("set private key", f"echo PRIVATE_KEY={priv_key} > account-abstraction/.env")
+        # set rpc url for deployment
+        lib.run("set rpc url", f"echo RPC_URL={config.l2_engine_rpc} >> account-abstraction/.env")
+        log_file = "logs/deploy_4337_contracts.log"
+        lib.run_roll_log(
+            "deploy contracts", 
+            command=deps.cmd_with_node("yarn deploy --network opstack"), 
+            cwd="account-abstraction",
+            log_file=log_file
+        )
+        print("Account abstraction contracts successfully deployed.")
+    else:
+        print("Account abstraction contracts already deployed.")
+
+def setup_stackup_bundler(config: L2Config):
+    github_url = "github.com/stackup-wallet/stackup-bundler"
+    version = "latest"
+
+    lib.run("installing stackup bundler", f"go install {github_url}@{version}")
+    print("Installation successful")
+
+    # set environment variables for bundler
+    lib.run(
+        "set full node RPC",
+        f"echo ERC4337_BUNDLER_ETH_CLIENT_URL={config.l2_engine_rpc} > .env"
+    )
+    if config.bundler_key is None:
+        priv_key = input("Enter private key for bundler: ")
+        config.bundler_key = priv_key
+    else:
+        priv_key = config.bundler_key
+    lib.run("set private key", f"echo ERC4337_BUNDLER_PRIVATE_KEY={priv_key} >> .env")
+
+    # make sure that GOPATH is set in PATH
+    current_path = os.environ.get("PATH", "")
+    gopath = subprocess.check_output(["go", "env", "GOPATH"]).decode().strip()
+    # append the bin directory of GOPATH to PATH
+    new_path = f"{gopath}/bin:{current_path}"
+    os.environ["PATH"] = new_path
```

Comment:
This should probably be done upstream (in `deps.py` / `check_go`) to avoid duplication and surprising PATH changes.

Do mention there that this is needed for the paymaster since I don't think we need it for the rest of the Go stuff.

Reply:
Done!